compiler/rustc_codegen_llvm/src/base.rs

```diff
@@ -97,21 +97,23 @@ pub fn compile_codegen_unit(
     tcx: TyCtxt<'tcx>,
     cgu_name: Symbol,
 ) -> (ModuleCodegen<ModuleLlvm>, u64) {
-    let prof_timer = tcx.prof.generic_activity_with_arg("codegen_module", cgu_name.to_string());
     let start_time = Instant::now();
 
     let dep_node = tcx.codegen_unit(cgu_name).codegen_dep_node(tcx);
     let (module, _) =
         tcx.dep_graph.with_task(dep_node, tcx, cgu_name, module_codegen, dep_graph::hash_result);
     let time_to_codegen = start_time.elapsed();
-    drop(prof_timer);
 
     // We assume that the cost to run LLVM on a CGU is proportional to
     // the time we needed for codegenning it.
     let cost = time_to_codegen.as_nanos() as u64;
 
     fn module_codegen(tcx: TyCtxt<'_>, cgu_name: Symbol) -> ModuleCodegen<ModuleLlvm> {
         let cgu = tcx.codegen_unit(cgu_name);
+        let _prof_timer = tcx.prof.generic_activity_with_args(
+            "codegen_module",
+            &[cgu_name.to_string(), cgu.size_estimate().to_string()],
+        );
```

Comment:
It looks like this now excludes the codegen_unit query -- is that intentional? I guess the query was always pre-executed which is why results are not affected?

Reply:
The `codegen_unit` query was already invoked just before running the task of `module_codegen`.

Reply:
Yes, but previously that was included in the timer here I think?

Reply:
Yeah, the query is certainly run before this point so the only difference is the fast query lookup and the fact the query is run inside this activity instead of right before it. I don't think moving the query out of the activity is significant so that seemed fine to me.

Reply:
Ok, makes sense. Thanks!